src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -250,10 +250,14 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
 
     // Don't allow a large zoom if the graph has a few nodes (nodes would look too big).
     // TODO: only do this if there is a small number of nodes - let the user zoom in large graphs
-    if (cy.zoom() > 2.5) {
-      cy.zoom(2.5);
+    // TODO: Remove setTimeout. It was added because zoom adjustment is not working fine with Cose layout.
+    setTimeout(() => {
```

Comment:
Cytoscape layout have events, one is triggered when the layout starts running, one when all the nodes are in position and one when the layout stops. I think we can hook on that. There is currently a PR open (#393) that does some changes on when this code is called.
Can you put a follow after that one is merged? or if you want, i can put it, i have a rough idea of what needs to be done, instead of the setTimeout i would suggest a [pon](http://js.cytoscape.org/#cy.promiseOn) or [one](http://js.cytoscape.org/#cy.one) when is really [needed](https://github.com/kiali/kiali-ui/pull/393/files#diff-04b15b4602d6b9ccf1eb82d15e07535fR231).